src/util/std_types.h

```diff
@@ -999,6 +999,13 @@ inline code_typet &to_code_type(typet &type)
   return static_cast<code_typet &>(type);
 }
 
+template <>
+inline bool can_cast_type<code_typet>(const typet &type)
+{
+  return type.id() == ID_code;
```

Comment:
When I added one, I used it as the pre-condition for `to_`  methods above - this way if we want to add more checks (e.g. that a `code_typet` always has a return type for example, there is only one place to change, what do you think?

Reply:
I've pushed an update. Let me know if that's what you meant.